code/controllers/voting.dm

```diff
@@ -11,385 +17,366 @@ datum/controller/vote
+			world << "<font color='red'><b>The round will start soon.</b></font>"
+
+	if(restart)
+		world << "World restarting due to vote..."
+		feedback_set_details("end_error", "restart vote")
+		if(blackbox)
+			blackbox.save_all_data_to_sql()
+		sleep(50)
+		log_game("Rebooting due to restart vote")
+		world.Reboot()
+
+/datum/controller/vote/proc/submit_vote(var/ckey, var/newVote)
+	if(mode)
+		if(config.vote_no_dead && usr.stat == DEAD && !usr.client.holder)
+			return
+		if(current_votes[ckey])
+			choices[choices[current_votes[ckey]]]--
+		if(newVote && newVote >= 1 && newVote <= choices.len)
+			choices[choices[newVote]]++
+			current_votes[ckey] = newVote
+		else
+			current_votes[ckey] = null
+
+/datum/controller/vote/proc/initiate_vote(var/vote_type, var/initiator_key, var/automatic = 0)
+	if(!mode)
+		if(started_time != null && !(check_rights(R_ADMIN) || automatic))
+			var/next_allowed_time = (started_time + config.vote_delay)
+			if(next_allowed_time > world.time)
 				return 0
-			if(vote && vote >= 1 && vote <= choices.len)
-				if(current_votes[ckey])
-					choices[choices[current_votes[ckey]]]--
-				voted += usr.ckey
-				choices[choices[vote]]++	//check this
-				current_votes[ckey] = vote
-				return vote
-		return 0
-
-	proc/initiate_vote(var/vote_type, var/initiator_key, var/automatic = 0)
-		if(!mode)
-			if(started_time != null && !(check_rights(R_ADMIN) || automatic))
-				var/next_allowed_time = (started_time + config.vote_delay)
-				if(next_allowed_time > world.time)
-					return 0
 
-			reset()
-			switch(vote_type)
-				if("restart")
-					choices.Add("Restart Round","Continue Playing")
-				if("gamemode")
-					if(ticker.current_state >= 2)
+		reset()
+
+		switch(vote_type)
+			if(VOTE_RESTART)
+				choices.Add("Restart Round", "Continue Playing")
+			if(VOTE_GAMEMODE)
+				if(ticker.current_state >= GAME_STATE_SETTING_UP)
+					return 0
+				choices.Add(config.votable_modes)
+				for(var/F in choices)
+					var/datum/game_mode/M = gamemode_cache[F]
+					if(!M)
+						continue
+					gamemode_names[M.config_tag] = capitalize(M.name) //It's ugly to put this here but it works
+					additional_text.Add("<td align = 'center'>[M.required_players]</td>")
+				gamemode_names["secret"] = "Secret"
+			if(VOTE_CREW_TRANSFER)
+				if(!check_rights(R_ADMIN|R_MOD, 0)) // The gods care not for the affairs of the mortals
+					if(get_security_level() == "red" || get_security_level() == "delta")
+						initiator_key << "The current alert status is too high to call for a crew transfer!"
 						return 0
-					choices.Add(config.votable_modes)
-					for (var/F in choices)
-						var/datum/game_mode/M = gamemode_cache[F]
-						if(!M)
-							continue
-						gamemode_names[M.config_tag] = capitalize(M.name) //It's ugly to put this here but it works
-						additional_text.Add("<td align = 'center'>[M.required_players]</td>")
-					gamemode_names["secret"] = "Secret"
-				if("crew_transfer")
-					if(check_rights(R_ADMIN|R_MOD, 0))

```

Comment:
Like, literally, I've changed  the check order to remove duplicated code but it's an exactly same check.